as/plugins/org.jboss.ide.eclipse.as.ui/jbossui/org/jboss/ide/eclipse/as/ui/views/server/extensions/ShowInWelcomePageActionProvider.java

```diff
@@ -175,8 +178,19 @@ private String getUrl() throws CoreException {
 			} else {
 				// When no module is selected,use welcome page url
 				ServerExtendedProperties props = (ServerExtendedProperties)server.loadAdapter(ServerExtendedProperties.class, new NullProgressMonitor());
-				if( props != null )
-					urlString = props.getWelcomePageUrl();
+				if( props != null ) {
+					try {
+						urlString = props.getWelcomePageUrl();
+					} catch (GetWelcomePageURLException e) {
+						Display.getDefault().asyncExec(new Runnable() {
+							public void run() {
+								MessageDialog.openWarning(
+										PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell(), 
+										"No Welcome URL to Show", e.getMessage());
```

Comment:
I believe the action is now called Show-in -> Web Browser,  not welcome page.  The welcome page terms are legacy and hidden.  Can we change this message to be more generic?  No URL found for current selection"  or something like that?